### PR TITLE
feature(CLI): Analyzer Resource Integration

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the actions and formatter to support the analyzer resource from the CLI

## Changes

- Adds analyzer actions
- Adds analyzer formatter
- Registers Analyzer actions to the resource actions object

## Fixes

- #2722 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

https://www.loom.com/share/45983054b6c548b6a21ed8b9a041936a

### CODE REVIEW

**Review by kdhamric — APPROVED** [review]
Watched the loom - looked great!

File: cli/actions/analyzer.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		return "", err
+	}
+
+	return *resource.(openapi.LinterResource).Spec.Id, nil
+}
+
+func (analyzer analyzerActions) Apply(ctx context.Context, fileContent file.File) (result *file.File, err error) {
+	result, err = analyzer.resourceClient.Update(ctx, fileContent, currentConfigID)
+	return result, err
+}
```
Comment (danielbdias) [inline cli/actions/analyzer.go:46]:
```suggestion
func (analyzer analyzerActions) Apply(ctx context.Context, fileContent file.File) (*file.File, error) {
  return analyzer.resourceClient.Update(ctx, fileContent, currentConfigID)
}
```

File: testing/cli-e2etest/testscenarios/analyzer/get_config_test.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	"github.com/kubeshop/tracetest/cli-e2etest/testscenarios/types"
+	"github.com/kubeshop/tracetest/cli-e2etest/tracetestcli"
+	"github.com/stretchr/testify/require"
+)
+
+func addGetAnalyzerPreReqs(t *testing.T, env environment.Manager) {
+	cliConfig := env.GetCLIConfigPath(t)
+
+	// When I try to set up a config
+	// Then it should be applied with success
```
Comment (danielbdias) [inline testing/cli-e2etest/testscenarios/analyzer/get_config_test.go:19]:
```suggestion
	// When I try to set up an analyzer
	// Then it should be applied with success
```

File: testing/cli-e2etest/testscenarios/analyzer/get_config_test.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+	env := environment.CreateAndStart(t)
+	defer env.Close(t)
+
+	cliConfig := env.GetCLIConfigPath(t)
+
+	t.Run("get with no analyzer initialized", func(t *testing.T) {
+		// Given I am a Tracetest CLI user
+		// And I have my server recently created
+		// And no config previously registered
```
Comment (danielbdias) [inline testing/cli-e2etest/testscenarios/analyzer/get_config_test.go:38]:
```suggestion
		// And no analyzer previously registered
```

File: testing/cli-e2etest/testscenarios/analyzer/get_config_test.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+		// When I try to get a config on yaml mode
```
Comment (danielbdias) [inline testing/cli-e2etest/testscenarios/analyzer/get_config_test.go:40]:
```suggestion
		// When I try to get an analyzer on yaml mode
```

File: testing/cli-e2etest/testscenarios/analyzer/get_config_test.go [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+	})
+
+	addGetAnalyzerPreReqs(t, env)
+
+	t.Run("get with YAML format", func(t *testing.T) {
+		// Given I am a Tracetest CLI user
+		// And I have my server recently created
+		// And I have a config already set
+
+		// When I try to get a config on yaml mode
```
Comment (danielbdias) [inline testing/cli-e2etest/testscenarios/analyzer/get_config_test.go:61]:
```suggestion
		// And I have an analyzer already set

		// When I try to get a analyzer on yaml mode
```

File: testing/cli-e2etest/testscenarios/analyzer/get_config_test.go [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+		require.Equal(config.Spec.MinimumScore, 95)
+		require.Len(config.Spec.Plugins, 3)
+	})
+
+	t.Run("get with JSON format", func(t *testing.T) {
+		// Given I am a Tracetest CLI user
+		// And I have my server recently created
+		// And I have a config already set
+
+		// When I try to get a config on json mode
```
Comment (danielbdias) [inline testing/cli-e2etest/testscenarios/analyzer/get_config_test.go:80]:
```suggestion
		// And I have an analyzer already set

		// When I try to get an analyzer on json mode
```

File: testing/cli-e2etest/testscenarios/analyzer/get_config_test.go [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+		require.Len(config.Spec.Plugins, 3)
+	})
+
+	t.Run("get with pretty format", func(t *testing.T) {
+		// Given I am a Tracetest CLI user
+		// And I have my server recently created
+		// And I have a config already set
+
+		// When I try to get a config on pretty mode
+		// Then it should print a table with 4 lines printed: header, separator, a config item and empty line
```
Comment (danielbdias) [inline testing/cli-e2etest/testscenarios/analyzer/get_config_test.go:100]:
```suggestion
		// And I have an analyzer already set

		// When I try to get an analyzer on pretty mode
		// Then it should print a table with 4 lines printed: header, separator, an analyzer item and empty line
```

File: testing/cli-e2etest/testscenarios/analyzer/list_config_test.go [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+	"github.com/kubeshop/tracetest/cli-e2etest/helpers"
+	"github.com/kubeshop/tracetest/cli-e2etest/testscenarios/types"
+	"github.com/kubeshop/tracetest/cli-e2etest/tracetestcli"
+	"github.com/stretchr/testify/require"
+)
+
+func addListAnalyzerPreReqs(t *testing.T, env environment.Manager) {
+	cliConfig := env.GetCLIConfigPath(t)
+
+	// When I try to set up a new config
```
Comment (danielbdias) [inline testing/cli-e2etest/testscenarios/analyzer/list_config_test.go:18]:
```suggestion
	// When I try to set up a new analyzer
```

File: testing/cli-e2etest/testscenarios/analyzer/list_config_test.go [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+	cliConfig := env.GetCLIConfigPath(t)
+
+	t.Run("list with no analyzer initialized", func(t *testing.T) {
+		// Given I am a Tracetest CLI user
+		// And I have my server recently created
+
+		// When I try to list config on pretty mode and there is no config previously registered
+		// Then it should print an empty table
+		// Then it should print a table with 4 lines printed: header, separator, the default config item and empty line
```
Comment (danielbdias) [inline testing/cli-e2etest/testscenarios/analyzer/list_config_test.go:41]:
```suggestion
		// When I try to list an analyzer on pretty mode and there is no analyzer previously registered
		// Then it should print an empty table
		// Then it should print a table with 4 lines printed: header, separator, the default analyzer item and empty line
```

File: testing/cli-e2etest/testscenarios/analyzer/list_config_test.go [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+	})
+
+	addListAnalyzerPreReqs(t, env)
+
+	t.Run("list with invalid sortBy field", func(t *testing.T) {
+		// Given I am a Tracetest CLI user
+		// And I have my server recently created
+		// And I already have a config created
+
+		// When I try to list a config by an invalid field
```
Comment (danielbdias) [inline testing/cli-e2etest/testscenarios/analyzer/list_config_test.go:57]:
```suggestion
		// And I already have an analyzer created

		// When I try to list one analyzer by an invalid field
```

File: testing/cli-e2etest/testscenarios/analyzer/list_config_test.go [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+		helpers.RequireExitCodeEqual(t, result, 1)
+		require.Contains(result.StdErr, "invalid sort field: id") // TODO: think on how to improve this error handling
+	})
+
+	t.Run("list with YAML format", func(t *testing.T) {
+		// Given I am a Tracetest CLI user
+		// And I have my server recently created
+		// And I already have a config created
+
+		// When I try to list config again on yaml mode
```
Comment (danielbdias) [inline testing/cli-e2etest/testscenarios/analyzer/list_config_test.go:69]:
```suggestion
		// And I already have an analyzer created

		// When I try to list one analyzer again on yaml mode
```

File: testing/cli-e2etest/testscenarios/analyzer/list_config_test.go [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+		require.Equal(configYAML[0].Spec.MinimumScore, 95)
+		require.Len(configYAML[0].Spec.Plugins, 3)
+	})
+
+	t.Run("list with JSON format", func(t *testing.T) {
+		// Given I am a Tracetest CLI user
+		// And I have my server recently created
+		// And I already have a config created
+
+		// When I try to list config again on json mode
```
Comment (danielbdias) [inline testing/cli-e2etest/testscenarios/analyzer/list_config_test.go:90]:
```suggestion
		// And I already have an analyzer created

		// When I try to list one analyzer again on json mode
```

File: testing/cli-e2etest/testscenarios/analyzer/list_config_test.go [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+		require.Len(configYAML[0].Spec.Plugins, 3)
+	})
+
+	t.Run("list with pretty format", func(t *testing.T) {
+		// Given I am a Tracetest CLI user
+		// And I have my server recently created
+		// And I already have a config created
+
+		// When I try to list config again on pretty mode
+		// Then it should print a table with 4 lines printed: header, separator, config item and empty line
```
Comment (danielbdias) [inline testing/cli-e2etest/testscenarios/analyzer/list_config_test.go:112]:
```suggestion
		// And I already have an analyzer created

		// When I try to list one analyzer again on pretty mode
		// Then it should print a table with 4 lines printed: header, separator, an analyzer item and empty line
```

**Review by danielbdias — APPROVED** [review]
🤘

File: testing/cli-e2etest/testscenarios/analyzer/delete_config_test.go [hunk 14]
```diff
@@ -0,0 +1,30 @@
+package analyzer
```
Comment (jorgeepc) [inline testing/cli-e2etest/testscenarios/analyzer/delete_config_test.go:1]: should the name of this file be `delete_analyzer_test.go`?

**Review by jorgeepc — APPROVED** [review]
Looks good, a couple of questions.

**Pull request merged**